qiskit/chemistry/ground_state_calculation/mes_factory.py

```diff
@@ -0,0 +1,60 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The minimum eigensolver factory for ground state calculation algorithms."""
+
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import MinimumEigensolver, VQE
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.qubit_transformations import QubitOperatorTransformation
+from qiskit.chemistry.components.initial_states import HartreeFock
+
+
+class MESFactory:
```

Comment:
Name should reflect that this takes a `Transformation` as input and returns a MES.

Reply:
Naming is to be discussed.